homeassistant/components/light/aurora.py

```diff
@@ -0,0 +1,158 @@
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_COLOR_TEMP, ATTR_EFFECT, ATTR_HS_COLOR,
+    SUPPORT_EFFECT, SUPPORT_BRIGHTNESS, SUPPORT_COLOR_TEMP,
+    SUPPORT_COLOR, PLATFORM_SCHEMA, Light)
+from homeassistant.const import CONF_HOST, CONF_TOKEN, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import color as color_util
+from homeassistant.util.color import \
+    color_temperature_mired_to_kelvin as mired_to_kelvin
+
+REQUIREMENTS = ['nanoleaf==0.4.1']
+
+SUPPORT_AURORA = (SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP | SUPPORT_EFFECT |
+                  SUPPORT_COLOR)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Optional(CONF_NAME, default='Aurora'): cv.string,
+})
+
+
+def aurora():
+    """Return the Nanoleaf module."""
+    import nanoleaf as nanoleaf_module
+    return nanoleaf_module
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup Nanoleaf Aurora device."""
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    token = config.get(CONF_TOKEN)
+    aurora_light = aurora().Aurora(host, token)
+    aurora_light.hass_name = name
+
+    if aurora_light.on is None:
+        _LOGGER.error("Could not connect to \
+        Nanoleaf Aurora: %s on %s", name, host)
+    add_devices([AuroraLight(aurora_light)])
+
+
+class AuroraLight(Light):
+    """Representation of a Nanoleaf Aurora."""
+
+    def __init__(self, light):
+        """Initialize an Aurora."""
+        self._brightness = light.brightness
+        self._color_temp = light.color_temperature
+        self._effect = light.effect
+        self._effects_list = light.effects_list
+        self._light = light
+        self._name = light.hass_name
+        self._hs_color = light.hue, light.saturation
+        self._state = light.on
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        if self._brightness is not None:
+            return int(self._brightness * 2.55)
+        return None
+
+    @property
+    def color_temp(self):
+        """Return the current color temperature."""
+        if self._color_temp is not None:
+            return color_util.color_temperature_kelvin_to_mired(
+                self._color_temp)
+        return None
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects."""
+        return self._effects_list
+
+    @property
+    def name(self):
+        """Return the display name of this light."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return "mdi:triangle-outline"
```

Comment:
I kinda like it since the device itself is a triangle, however this is more of a special snowflake kind of situation, so if you'd like to keep this as a regular light bulb I'll happily oblige. 

It looks like this (I hope it's okay to attach it here, I don't think it fits into the documentation itself):
![screen shot 2018-04-04 at 15 36 19](https://user-images.githubusercontent.com/357005/38310874-f5345dc0-381d-11e8-8e2e-3167ff9c6b4d.png)

Reply:
👍 Looks nice to me, and these aren't normal lights, so a different icon seems reasonable to me.

![image](https://user-images.githubusercontent.com/1386547/38311263-c213ffb2-37ec-11e8-96b7-f59f263843a7.png)